### ISSUE TITLE
Upgrage MGCB fallback version

### ISSUE THREAD

**Issue by spaaaaam** [infinitespace-studios/InfinitespaceStudios.Pipeline#1]
Hello,

following a long post [here](http://community.monogame.net/t/solved-shader-different-mgfx-binary-format-between-mono-and-net-core/11689/21), i think that the pipeline service fallbacks to MG 3.5 when it does not recognize the version of the client calling it ([source](https://github.com/infinitespace-studios/InfinitespaceStudios.Pipeline/blob/f9d93e9aedcbd5ecd56178471a5ddf89edd0737f/InfinitespaceStudios.PipelineService/PipelineRole/Controllers/EffectController.cs#L51)).

This appears in particular when using MG dotnet core build.

This is a problem since MG fx binary headers have changed since MG 3.6 ([source](https://github.com/infinitespace-studios/InfinitespaceStudios.Pipeline/blob/f9d93e9aedcbd5ecd56178471a5ddf89edd0737f/InfinitespaceStudios.PipelineService/PipelineRole/Controllers/EffectController.cs#L51)). Hence binary effects compiled with MG 3.5 are incompatible with recent MG nuget packages.

Would it be possible to bump the fallback version a bit?

Many thanks for this nice service by the way, which is a real lifeline for linux users!

**Comment by Jjagg** [comment]
It would be better if the local part of the plugin could reflect into the local MGCB assembly to get the shader format version instead of depending on the MG version. It could also take a parameter for the version to force a certain formar. On the MG side we should also make some improvements:

- we should add the MGFX version to the help screen in the Pipeline Tool and expose it from 2MGFX so it's more discoverable. 
- The error message MG throws when failing to load should mention what version the shader is and what's expected.
- host this plugin for the different versions of the shader format 
- use this implementation for Mac/Linux by default without a plugin

**Comment by dellis1972** [comment]
I just added version 3.7.1 to the service https://github.com/infinitespace-studios/InfinitespaceStudios.Pipeline/commit/65144ceed4ae1bfed8ce4c8eb2066baff170e668. 
That was probably what was causing the issue. If 3.7.1 was not found it would default back to 3.5. I also changed the way the code looks up the matching version. It will try to match on the full version first, then Major, Minor , Build  then just Major Minor. So if you send 3.6.1.45234 you will end up with the 3.6 version. That said, if you are using this service against the develop branch of MG, it won't work as those binaries are not in this system, it only supports the stable releases. So sending up 3.8.x.x will result in it falling back to 3.5.  

Another idea would be to auto download builds from team city so that many versions are supported. Rather than having to re-publish the service each time a new release is done. Problem with that is an ever expanding requirement for storage.

At some point I am going to have to shut this service down as its costing me to keep it running. If MonoGame could host it, that would be a much better solution in the long run. Ideally this should be built into MonoGame rather than a extension.
